Hmm this situation reproduces on `main` with the existing cohere unit test -- test_cohere_completion:

```

test_cohere_completion.py:20 (test_chat_completion_cohere)
def test_chat_completion_cohere():
        try:
            litellm.set_verbose = True
            messages = [
                {
                    "role": "user",
                    "content": "Hey",
                },
            ]
>           response = completion(
                model="cohere_chat/command-r",
                messages=messages,
                max_tokens=10,
            )

test_cohere_completion.py:30: 
```

```
                    extra_body[k] = passed_params[k]
            optional_params["extra_body"] = extra_body
        else:
            # if user passed in non-default kwargs for specific providers/models, pass them along
>           for k in passed_params.keys():
E           RuntimeError: dictionary changed size during iteration

../utils.py:5772: RuntimeError

During handling of the above exception, another exception occurred:

    def test_chat_completion_cohere_tool_calling():
        try:
            litellm.set_verbose = True
            messages = [
                {
                    "role": "user",
                    "content": "What is the weather like in Boston?",
                },
            ]
            response = completion(
                model="cohere_chat/command-r",
                messages=messages,
                tools=[
                    {
                        "type": "function",
                        "function": {
                            "name": "get_current_weather",
                            "description": "Get the current weather in a given location",
                            "parameters": {
                                "type": "object",
                                "properties": {
                                    "location": {
                                        "type": "string",
                                        "description": "The city and state, e.g. San Francisco, CA",
                                    },
                                    "unit": {
                                        "type": "string",
                                        "enum": ["celsius", "fahrenheit"],
                                    },
                                },
                                "required": ["location"],
                            },
                        },
                    }
                ],
            )
            print(response)
        except Exception as e:
>           pytest.fail(f"Error occurred: {e}")
E           Failed: Error occurred: dictionary changed size during iteration

test_cohere_completion.py:100: Failed
``` 

Do these get run in CI anywhere?

--- a/litellm/utils.py
+++ b/litellm/utils.py
@@ -4836,7 +4836,7 @@ def get_optional_params(
     **kwargs,
 ):
     # retrieve all parameters passed to the function
-    passed_params = locals()
+    passed_params = locals().copy()
     special_params = passed_params.pop("kwargs")
     for k, v in special_params.items():
         if k.startswith("aws_") and (
@@ -5770,7 +5770,7 @@ def get_optional_params(
         optional_params["extra_body"] = extra_body
     else:
         # if user passed in non-default kwargs for specific providers/models, pass them along
-        for k in list(passed_params.keys()):
+        for k in passed_params.keys():
             if k not in default_params.keys():
                 optional_params[k] = passed_params[k]
     print_verbose(f"Final returned optional params: {optional_params}")

